src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -73,6 +73,22 @@ void QgsMeshLayer::setDefaultRendererSettings()
   }
 }
 
+void QgsMeshLayer::createSimplifiedMeshes()
+{
+  if ( mSimplifySettings.isEnabled() && mTriangularMeshes.size() == 1 )
```

Comment:
`mTriangularMeshes.size() == 1` means there is only base triangular mesh and nothing else?
if yes, maybe create function `bool hasSimplifiedMeshes() const` for better readability